packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -965,10 +1003,13 @@ function processFields(
   fields: Record<string, any>,
   impliedFields: Record<string, any>,
   fieldLevelAuthRules: Record<string, string | null>,
+  getRefType: ReturnType<typeof getRefTypeForSchema>,
+
   identifier?: readonly string[],
   partitionKey?: string,
   secondaryIndexes: TransformedSecondaryIndexes = {},
   databaseEngine: DatasourceEngine = 'dynamodb',
+  isInput: boolean = false,
```

Comment:
What does `isInput` mean in this context? Can the name be clearer?

Even before this change, this function is taking enough defaulted params that it might be worth using a named args object instead of receiving this long list of different args with defaults that we can't use depending on which ordered arg we want to override.